### PR TITLE
Motorcycle controller

### LINKED ISSUE #10: [1xpt] Create Motor cycle Controller
### - Create An Api such as:

- [ ] To get a motor by an ID
- [ ] To get recently posted Motors
- [ ] Paginations

### PR DESCRIPTION
**Pull request by karayamanemre**
Closes #10 
- Created API motorcycles controller with required actions.

### CODE REVIEW

File: app/controllers/api/motorcycles_controller.rb [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+  before_action :set_motorcycle, only: [:show, :update, :destroy]
+
+  def index
+    @motorcycles = Motorcycle.all
+    render json: @motorcycles
+  end
+
+  def show
+    render json: @motorcycle
+  end
```
Comment (NatanimA) [inline app/controllers/api/motorcycles_controller.rb]: Which motorcycle are we returning here, I think we first need to find which motor is being requested, do
`motor = Motorcycle.find_by(id:params[:id]) #then`
```
if not  motor.nil?
#statemetns
end
```
- And Add messages and status code in the response body

File: app/controllers/api/motorcycles_controller.rb [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+
+  def create
+    @motorcycle = Motorcycle.new(motorcycle_params)
+
+    if @motorcycle.save
+      render json: @motorcycle, status: :created, location: @motorcycle
+    else
+      render json: @motorcycle.errors, status: :unprocessable_entity
+    end
+  end
```
Comment (NatanimA) [inline app/controllers/api/motorcycles_controller.rb]: Here before saving validate is the created instance is correct. if not return a` status of 400 `with bad request as a message.

File: app/controllers/api/motorcycles_controller.rb [hunk 3]
```diff
@@ -1,2 +1,44 @@
...
+
+  def update
+    if @motorcycle.update(motorcycle_params)
+      render json: @motorcycle
+    else
+      render json: @motorcycle.errors, status: :unprocessable_entity
+    end
+  end
```
Comment (NatanimA) [inline app/controllers/api/motorcycles_controller.rb]: Handle exceptions that might happen, Such as validations errors,and send appropriate message and status accordingly.

File: app/controllers/api/motorcycles_controller.rb [hunk 4]
```diff
@@ -1,2 +1,44 @@
...
+
+  def destroy
+    @motorcycle.destroy
+    head :no_content
+  end
```
Comment (NatanimA) [inline app/controllers/api/motorcycles_controller.rb]: - First we need to check if motor exists, if not we will return a bad requset
- then after delete send a `status code of 200`

**Review by NatanimA — CHANGES_REQUESTED** [review]
Hi @karayamanemre 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Implemented Api controller for motorcycles ✔️ 

You can do it

## Required Changes ♻️

- [ ] Implement Exception handlings
- [ ] Implement validation



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: app/controllers/api/motorcycles_controller.rb (hunk 1 above)
Comment (karayamanemre) [reply app/controllers/api/motorcycles_controller.rb]: There is a private set_motorcycle method that retrieves the motorcycle with the given id parameter from the database and sets it to the @motorcycle instance variable. And Rails handles response status codes and messages automatically.

File: app/controllers/api/motorcycles_controller.rb (hunk 2 above)
Comment (karayamanemre) [reply app/controllers/api/motorcycles_controller.rb]: Done ✅

File: app/controllers/api/motorcycles_controller.rb (hunk 4 above)
Comment (karayamanemre) [reply app/controllers/api/motorcycles_controller.rb]: Added the check condition.

**Review by NatanimA — APPROVED** [review]
Hi @karayamanemre 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [x] Inserted Required changes✔️ 



Cheers and Happy coding!👏👏👏

**Pull request merged**